users/views.py

```diff
@@ -1,10 +1,76 @@
 from django.urls import reverse_lazy
+from django.http import HttpResponse
 from django.views import generic
+from django.core.mail import send_mail, BadHeaderError
+from django.shortcuts import render
+from django.contrib.auth import login
+from django.contrib.sites.shortcuts import get_current_site
+from django.utils.encoding import force_bytes, force_text
+from django.template.loader import render_to_string
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
 
+from .models import CustomUser
 from .forms import CustomUserCreationForm
+from .tokens import account_activation_token
 
 
 class SignUp(generic.CreateView):
     form_class = CustomUserCreationForm
     success_url = reverse_lazy("login")
     template_name = "signup.html"
+
+    def get(self, request):
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            # import ipdb
+
+            # ipdb.set_trace()
+            user = form.save(commit=False)
+            user.save()
+            current_site = get_current_site(request)
+            subject = "Activate your Bookx account."
+            message = render_to_string(
+                "acc_activate_email.html",
+                {
+                    "user": user,
+                    "domain": current_site.domain,
+                    "uid": urlsafe_base64_encode(
+                        force_bytes(user.pk)
+                    ).decode(),
+                    "token": account_activation_token.make_token(user),
```

Comment:
instead of passing domain and uid and token to the template, you could also build the url when building the context. There is a utility for composing absolute urls if you have a request object (like you do here):

https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.build_absolute_uri

so something like:

```python
{
    "activation_url": request.build_absolute_uri(
        reverse("activate", kwargs=dict(uidb64=uid token=token)),
    #...
}